src/common/settings.ts

```diff
@@ -0,0 +1,142 @@
+import { Dispatch, SetStateAction, useState } from "react";
+
+const defaultCommunityDir = (): string => {
+    // Ensure proper functionality in main- and renderer-process
+    const app = electronApp || remote.app;
+    let msfsConfigPath = null;
+
+    const steamPath = path.join(app.getPath('appData'), "\\Microsoft Flight Simulator\\UserCfg.opt");
+    const storePath = path.join(app.getPath('home'), "\\AppData\\Local\\Packages\\Microsoft.FlightSimulator_8wekyb3d8bbwe\\LocalCache\\UserCfg.opt");
+
+    if (fs.existsSync(steamPath)) {
+        msfsConfigPath = steamPath;
+    } else if (fs.existsSync(storePath)) {
+        msfsConfigPath = storePath;
+    } else {
+        walk(path.join(app.getPath('home'), "\\AppData\\Local\\"), (path) => {
+            if (path.includes("Flight") && path.includes("UserCfg.opt")) {
+                msfsConfigPath = path;
+            }
+        });
+    }
+
+    if (!msfsConfigPath) {
+        return 'C:\\';
+    }
+
+    const msfsConfig = fs.readFileSync(msfsConfigPath).toString();
+    const msfsConfigLines = msfsConfig.split(/\r?\n/);
+    const packagesPathLine = msfsConfigLines.find(line => line.includes('InstalledPackagesPath'));
+    const communityDir = path.join(packagesPathLine.split(" ").slice(1).join(" ").replaceAll('"', ''), "\\Community");
+
+    return fs.existsSync(communityDir) ? communityDir : 'C:\\';
+};
+
+export const persistWindowSettings = (window: Electron.BrowserWindow): void => {
+    store.set('cache.main.maximized', window.isMaximized());
+
+    const winSize = window.getSize();
+    store.set('cache.main.lastWindowX', winSize[0]);
+    store.set('cache.main.lastWindowY', winSize[1]);
+};
+
+export const useSetting = <T>(key: string): [T, Dispatch<SetStateAction<T>>] => {
+    const [storedValue, setStoredValue] = useState(store.get<string, T>(key));
+
+    store.onDidChange(key as never, (val) => {
+        setStoredValue(val as T);
+    });
+
+    const setValue = (newVal: T) => {
+        store.set(key, newVal);
+    };
+
+    return [storedValue, setValue];
+};
+
+const schema: Schema<unknown> = {
+    mainSettings: {
+        type: "object",
+        // Empty defaults are required when using type: "object" (https://github.com/sindresorhus/conf/issues/85#issuecomment-531651424)
+        default: {},
+        properties: {
+            separateLiveriesPath: {
+                type: "boolean",
+                default: false
+            },
+            disableExperimentalWarning: {
+                type: "boolean",
+                default: false,
+            },
+            disabledIncompatibleLiveriesWarning: {
+                type: "boolean",
+                default: false,
+            },
+            useCdnCache: {
+                type: "boolean",
+                default: true,
+            },
+            dateLayout: {
+                type: "string",
+                default: "yyyy/mm/dd"
+            },
+            msfsPackagePath: {
```

Comment:
I think we should rename this to communityPath for better understanding and add a condition in defaultCommunityDir() that adopts the old setting

Reply:
We can use migrations now in the newly introduced schema that can take care of that.

Reply:
Not sure if we should change that since it isn't necessarily the community directory.

Reply:
well in Directories we pretend it is. I get your point, but I think we should have a common naming for it

Reply:
I'm all for changing it in Directories then. I don't want to migrate the settings just yet.

Reply:
then leave it as it is rn